### DISCUSSION TITLE
How to load .env variables in my zx script?

### DISCUSSION THREAD

**Discussion by codenoobforreal** [google/zx#792] [Help needed] [answered]
let's say I have a .env file in my project root and I want to get my environment variables.

**Comment by antongolub** [comment, marked as answer]
zx uses [`process.env` object by reference](https://github.com/google/zx/blob/main/src/core.ts#L97), so you are able to modify this value in any convenient way. For example, with [dotenv](https://www.npmjs.com/package/dotenv).

**Reply by antonmedv** [reply]
Should we include dotenv be default ?

**Reply by antongolub** [reply]
dotenv ~80Kb
dotenvx ~1Mb

It seems better to implement own lightweight version

**Reply by antonmedv** [reply]
> It seems better to implement own lightweight version

👍🏻

**Reply by codenoobforreal** [reply]
I know It's working in standalone repo after I tried to use zx in a vite project.
It needs one extra step when it comes to monorepo like turborepo.You must run script with `dotenv --` prefix(dotenv-cli installed) if your .env file is in the project root.
I was getting undefined when I run zx script without that prefix.So I asked for help in discussios.

**Comment by joshunger** [comment]
Node now has support for `.env` files so you don't have to use `dotenv` including `--env-file` and `process.loadEnvFile`

**Reply by antonmedv** [reply]
But we need to polyfill for older versions of node.
